test/type_prop.cpp

```diff
@@ -4343,12 +4343,32 @@ TEST(type_prop, one_hot_deduce_matrix_2)
     ASSERT_EQ(oh->get_shape(), (Shape{12, 24, 2}));
 }
 
+TEST(type_prop, one_hot_deduce_et_dynamic)
```

Comment:
> `one_hot_deduce_et_dynamic` 

Curious what the `_et` is referring to here: "element type" or maybe "element test" to determine dynamism?

Since one of the possible implementations for dynamic shapes is "track whether each node depends on dynamic shape information. If none of the results in a function depend on dynamic shapes, then the function does not." seems like this could be ambiguous enough to break graphs that *could* be handled dynamically.  I don't know what this means for this test, though.

Reply:
It stands for element type. I don’t understand the second paragraph of your comment. (Sorry, it’s early and no caffeine yet ;)) What is it that is ambiguous, and what do you mean by breaking graphs?

Reply:
Some of the drafts on the Intelpedia page for nGraph "Designs" mentioned the various ways that dynamic shapes and types might be implemented.  We also have some notes on how TensorFlow wants things to be done, but there might be other ways.  That's all

Reply:
Also, we haven't yet added any docs about dynamic graphs or shapes so I probably just don't yet understand these ops well enough.

Reply:
In any case I think this test should be okay :)